Pinta.Core/Managers/SettingsManager.cs

```diff
@@ -48,111 +54,106 @@ public SettingsManager ()
 		
 		public string GetUserSettingsDirectory ()
 		{
-			var settings_dir = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.ApplicationData), "Pinta");
+			if (settings_directory != null)
```

Comment:
I think this caching might run into issues if the user deletes the settings folder while Pinta is running, if code relies on the directory being created later in this method?